quartz/cfg.ts

```diff
@@ -55,6 +56,10 @@ export interface GlobalConfiguration {
    *   Quartz will avoid using this as much as possible and use relative URLs most of the time
    */
   baseUrl?: string
+  /**
+   * Wether to generate social images (Open Graph and Twitter standard) for link previews
+   */
+  generateSocialImages: boolean | Partial<SocialImageOptions>
```

Comment:
Thanks for the hard work. I think we can skip generating social image during `npx quartz build --serve`, to avoid generating a whole bunch of social images for larger vault.

Reply:
> Thanks for the hard work. I think we can skip generating social image during `npx quartz build --serve`, to avoid generating a whole bunch of social images for larger vault?
> 
> What do you think?

I don't think this is an issue for large vaults, as this feature needs to be manually enabled and is opt-in. Additionally, the generation of the images happens asynchronously, so startup performance shouldn't be impacted significantly. 

I would need to look at the details of my old implementation again to confirm this isn't one of the frontmatter attributes I already added, but the only thing I could see as being useful in this regard is setting a white/blacklist for certain pages (tho pretty sure I haven't added this currently, might be a future expansion as people shouldn't immediately be impacted from a feature that's opt-in and generates images that are already optimized and slightly compressed)